### PR TITLE
Issue: RuleSingleError should change ValidationStrategy in nested levels

### PR DESCRIPTION
**Pull request by Grosbo**
## Change
Changed:
-AsModelRule.cs and AsCollectionRule.cs

## Description
Simple "if" statement before executing validator  in both files.
Tested it out with simple unit test if it runs faster.
Im not sure if that was the expected result.

### CODE REVIEW

**Comment by bartoszlenar** [comment]
@Grosbo , thank you so much for participation in the project and for making this PR.

after investigation I discovered that the "Force" scenario is a little bit more complex than I first thought. more cases need to be covered considering messages (added using `WithMessage`) and I need to write additional unit tests to make sure it's all working as it should be.

would you mind to include only this part?

``` csharp
if (RuleSingleError != null && validationStrategy == ValidationStrategy.Complete)
{
      validationStrategy = ValidationStrategy.FailFast;
 }
``` 

this weekend I'm publishing version 1.0.0, so this could also be a part of the release.

**Comment by Grosbo** [comment]
You are welcome. I'm thankful to you for being friendly towards me, cause I hesitated to PR such a simple "if" statement and those are my first steps in real project on GitHub. I will try to get better understanding of your project and then tackle another issue.

File: src/CoreValidation/Specifications/Commands/AsCollectionRule.cs [hunk 1]
```diff
@@ -36,6 +36,13 @@ public override bool TryGetErrors(object model, object memberValue, IExecutionCo
 
         public bool TryGetErrors(TModel model, IEnumerable<TItem> memberValue, IExecutionContext executionContext, ValidationStrategy validationStrategy, int depth, out IErrorsCollection errorsCollection)
         {
+
+            if (RuleSingleError != null && validationStrategy == ValidationStrategy.Complete)
+            {
+                validationStrategy = ValidationStrategy.FailFast;
+
```
Comment (bartoszlenar) [inline src/CoreValidation/Specifications/Commands/AsCollectionRule.cs]: could you remove the whiteline?

File: src/CoreValidation/Specifications/Commands/AsModelRule.cs [hunk 2]
```diff
@@ -35,8 +35,15 @@ public bool TryGetErrors(TMember memberValue, IExecutionContext executionContext
         {
             var validator = executionContext.ValidatorsFactory.GetOrInit(Specification, SpecificationId);
 
+            if (RuleSingleError != null && validationStrategy == ValidationStrategy.Complete)
+            {
+                validationStrategy = ValidationStrategy.FailFast;
+
```
Comment (bartoszlenar) [inline src/CoreValidation/Specifications/Commands/AsModelRule.cs]: could you remove the whiteline?

File: src/CoreValidation/Specifications/Commands/AsCollectionRule.cs (hunk 1 above)
Comment (Grosbo) [reply src/CoreValidation/Specifications/Commands/AsCollectionRule.cs]: Sure! Sorry for that.

**Comment by bartoszlenar** [comment]
@Grosbo sorry for being quiet for a while. I'm on holidays and will be back next week.

all good, just one thing I forgot to ask you; please squash all the commits from the PR into a single one?

**Pull request closed without merging**